spec/jobs/reports/combined_invoice_supplement_report_v2_spec.rb

```diff
@@ -79,19 +102,52 @@
   let(:iaa2_range) { DateTime.new(2020, 9, 1).utc..DateTime.new(2021, 8, 30).utc }
   let(:inside_iaa2) { iaa2_range.begin + 1.day }
 
+  let(:iaa3) { 'iaa3' }
+  let(:iaa3_range) { DateTime.new(2020, 9, 1).utc..DateTime.new(2021, 8, 30).utc }
+  let(:inside_iaa3) { iaa3_range.begin + 2.days }
+
   describe '#perform' do
     it 'is empty with no data' do
       csv = CSV.parse(report.perform(Time.zone.today), headers: true)
       expect(csv).to be_empty
     end
 
-    context 'with data' do
-      let(:user1) { create(:user) }
-      let(:user2) { create(:user) }
-      let(:user3) { create(:user) }
-      let(:user4) { create(:user) }
-      let(:user5) { create(:user) }
-      let(:user6) { create(:user) }
+    context 'with data generates a report by iaa + order number, issuer and year month' do
+      let(:user1) { create(:user, profiles: [profile1]) }
+      let(:profile1) { build(:profile, verified_at: DateTime.new(2019, 10, 15).utc) }
+
+      let(:user2) { create(:user, profiles: [profile2]) }
+      let(:profile2) { build(:profile, verified_at: DateTime.new(2018, 6, 1).utc) }
+
+      let(:user3) { create(:user, profiles: [profile3]) }
+      let(:profile3) { build(:profile, verified_at: DateTime.new(2018, 6, 1).utc) }
+
+      let(:user4) { create(:user, profiles: [profile4]) }
+      let(:profile4) { build(:profile, verified_at: nil) }
+
+      let(:user5) { create(:user, profiles: [profile5]) }
+      let(:profile5) { build(:profile, verified_at: DateTime.new(2019, 1, 1).utc) }
+
+      let(:user6) { create(:user, profiles: [profile6]) }
+      let(:profile6) { build(:profile, verified_at: DateTime.new(2019, 1, 1).utc) }
+
+      let(:user7) { create(:user, profiles: [profile7]) }
+      let(:profile7) { build(:profile, verified_at: DateTime.new(2018, 1, 1).utc) }
+
+      let(:user8) { create(:user, profiles: [profile8]) }
+      let(:profile8) { build(:profile, verified_at: DateTime.new(2017, 1, 1).utc) }
+
+      let(:user9) { create(:user, profiles: [profile9]) }
+      let(:profile9) { build(:profile, verified_at: DateTime.new(2016, 1, 1).utc) }
+
+      let(:user10) { create(:user, profiles: [profile10]) }
+      let(:profile10) { build(:profile, verified_at: DateTime.new(2015, 1, 1).utc) }
+
+      let(:user11) { create(:user, profiles: [profile11]) }
+      let(:profile11) { build(:profile, verified_at: DateTime.new(2019, 10, 10).utc) }
+
+      let(:user12) { create(:user, profiles: [profile12]) }
+      let(:profile12) { build(:profile, verified_at: DateTime.new(2017, 9, 10).utc) }
 
       before do
```

Comment:
Just to quick check my understanding, looks this PR keeps the one shared `before` block, and adds additional test cases to it?

Reply:
Yeah, some users are reused in different it blocks but some aren't, we can clean that up a bit. Luis wanted to see if we could modularize it a bit before officially submitting the PR, which might make it cleaner

Reply:
Okay, this has been split up into smaller contexts with separate before blocks.  We certainly could modularize it even more, but this is how far we got in the timebox given. 👍 ⏳